### PR TITLE
Update Postgres streaming replication instructions

### LINKED ISSUE #2400: Update Postgres SR content
### Description
We will need to update the Postgres SR content to reflect the clarifications Edgar identified during testing.

### Category

This is an issue with:

- [ ] errata (error in doc content)
- [ ] new doc (requires a completely new doc)
- [x] update (add missing or refresh existing content)
- [x] improvement (improve docs content or format)

### Affected pages
https://docs.sensu.io/sensu-go/latest/guides/scale-event-storage/#configure-postgres-streaming-replication

### Corrections needed
Need to identify changes based on notes at https://docs.google.com/document/d/1ZN33pSWHfkhWKWwF55dCCpaw3DMvkmERfiHe2OpkVQo/edit

### PR DESCRIPTION
**Pull request by hillaryfraley**
## Description
Updates the instructions in https://docs.sensu.io/sensu-go/latest/operations/deploy-sensu/scale-event-storage/#configure-postgres-streaming-replication

## Motivation and Context
Closes https://github.com/sensu/sensu-docs/issues/2400

### CODE REVIEW

**Comment by hillaryfraley** [comment]
I'm working from https://github.com/sensu/skunkworks/pull/81 and updating according to notes from Mike as I go. Here's a summary of the alterations I'm using b/c the skunkworks scenario uses postgresql 9.6:
- postgresql-9.6 instead of just postgresql in all commands that mention it
- paths to the Postgres configuration files should use `/var/lib/pgsql/9.6/data`

~To do~Done: make it clearer that steps 1 and 2 should be on the “primary” postgres box (postgres-1 if using the skunkworks scenario) and step 3 should be on the “standby” postgres box (postgres-2 in scenario)

Right now, everything seems to work until I get toward the end of [step 3](https://docs.sensu.io/sensu-go/latest/operations/deploy-sensu/scale-event-storage/#step-3-bootstrap-the-standby-host). Here's everything starting in step 3:

```
hillaryfraley@Hillarys-Sensu-MacBook-Pro sensu-go-postgres-vagrant % vagrant ssh postgres-2
[vagrant@postgres-2 ~]$ sudo systemctl stop postgresql
Failed to stop postgresql.service: Unit postgresql.service not loaded.
[vagrant@postgres-2 ~]$ sudo mv /var/lib/pgsql/9.6/data /var/lib/pgsql/9.6/data.bak
[vagrant@postgres-2 ~]$ sudo install -d -o postgres -g postgres -m 0700 /var/lib/pgsql/9.6/data
[vagrant@postgres-2 ~]$ export PRIMARY_IP=10.0.10.1
[vagrant@postgres-2 ~]$ sudo -u postgres pg_basebackup -h $PRIMARY_IP -D /var/lib/pgsql/9.6/data -P -U repl -R --xlog-method=stream
could not change directory to "/home/vagrant": Permission denied
pg_basebackup: could not connect to server: Connection refused
	Is the server running on host "10.0.10.1" and accepting
	TCP/IP connections on port 5432?
```

File: content/sensu-go/6.4/operations/deploy-sensu/scale-event-storage.md [hunk 1]
```diff
@@ -435,46 +421,38 @@ After you enter your password, Postgres will list database copy progress:
...
-
+3. Check the commit log location from the standby host:
 {{< code shell >}}
 sudo -u postgres psql -c "select pg_last_xlog_receive_location()"
 sudo -u postgres psql -c "select pg_last_xlog_replay_location()"
 {{< /code >}}
 
-Postgres will list the commit log location:
-
+   Postgres will list the standby host commit log location:
```
Comment (hillaryfraley) [inline content/sensu-go/6.4/operations/deploy-sensu/scale-event-storage.md:455]: I think something might be missing here -- perhaps users need to confirm identical responses for the primary and standby hosts log locations? As written, the instructions don't explain how the steps here result in a confirmation that SR is working.

File: content/sensu-go/6.4/operations/deploy-sensu/scale-event-storage.md [hunk 2]
```diff
@@ -435,46 +421,38 @@ After you enter your password, Postgres will list database copy progress:
 30318/30318 kB (100%), 1/1 tablespace
 {{< /code >}}
 
-### Step 4: Confirm replication
-
-To confirm your configuration is working properly, first remove configurations that are only for the primary:
+### Confirm replication
 
+1. On the **primary** Postgres host, remove primary-only configurations:
```
Comment (hillaryfraley) [inline content/sensu-go/6.4/operations/deploy-sensu/scale-event-storage.md:426]: Need to confirm that this step should indeed be completed on the primary host. Does it matter which host users run this command from?

**Comment by hillaryfraley** [comment]
Working through the SR instructions again with the updated skunkworks scenario (now using Psql 12) -- still get the error at the end of step 3:

```
[vagrant@postgres-2 ~]$ sudo -u postgres pg_basebackup -h $PRIMARY_IP -D /var/lib/pgsql/12/data -P -U repl -R --xlog-method=stream
could not change directory to "/home/vagrant": Permission denied
pg_basebackup: unrecognized option '--xlog-method=stream'
Try "pg_basebackup --help" for more information.
```

edit: working w/Mike on this, we think xlog-method should be changed to wal-method (see https://www.postgresql.org/docs/12/app-pgbasebackup.html)

**Comment by hillaryfraley** [comment]
Review note: if you use the [new Postgres skunkworks scenario](https://github.com/sensu/skunkworks/pull/81) to test, here are the modified commands (you'll need to specify postgres version 12 in many of them, and two require the postgres host IPs):

```
sudo cp /var/lib/pgsql/12/data/pg_hba.conf /var/tmp/pg_hba.conf.bak

export STANDBY_IP=10.0.10.5

echo "host replication repl ${STANDBY_IP}/32 md5" | sudo tee -a /var/lib/pgsql/12/data/pg_hba.conf

sudo systemctl restart postgresql-12

sudo cp -a /var/lib/pgsql/12/data/postgresql.conf /var/lib/pgsql/12/data/postgresql.conf.bak

echo 'wal_level = replica' | sudo tee -a /var/lib/pgsql/12/data/postgresql.conf

echo 'max_wal_senders = 5' | sudo tee -a /var/lib/pgsql/12/data/postgresql.conf

echo 'wal_keep_segments = 32' | sudo tee -a /var/lib/pgsql/12/data/postgresql.conf

sudo systemctl restart postgresql-12

sudo systemctl stop postgresql-12

sudo mv /var/lib/pgsql/12/data /var/lib/pgsql/12/data.bak

sudo install -d -o postgres -g postgres -m 0700 /var/lib/pgsql/12/data

export PRIMARY_IP=10.0.10.4

sudo -u postgres pg_basebackup -h $PRIMARY_IP -D /var/lib/pgsql/12/data -P -U repl -R --wal-method=stream

sudo sed -r -i.bak '/^(wal_level|max_wal_senders|wal_keep_segments).*/d' /var/lib/pgsql/12/data/postgresql.conf

sudo systemctl start postgresql-12
```

**Comment by lspxxv** [comment]
The revised steps work as expected. I have a minor issue with etcd on a sensu backend node but a vagrant destroy and recreation of backend-1 did the trick.

**Pull request merged**